src/cargo/ops/cargo_rustc/output_depinfo.rs

```diff
@@ -79,15 +79,31 @@ pub fn output_depinfo<'a, 'b>(context: &mut Context<'a, 'b>, unit: &Unit<'a>) ->
         }
         None => None,
     };
+    let deps = deps.iter()
+        .map(|f| render_filename(f, basedir))
+        .collect::<CargoResult<Vec<_>>>()?;
+
     for &(_, ref link_dst, _) in context.target_filenames(unit)?.iter() {
         if let Some(ref link_dst) = *link_dst {
             let output_path = link_dst.with_extension("d");
             if success {
-                let mut outfile = BufWriter::new(File::create(output_path)?);
                 let target_fn = render_filename(link_dst, basedir)?;
+
+                // If nothing changed don't recreate the file which could alter
+                // its mtime
+                if let Ok(previous) = fingerprint::parse_rustc_dep_info(&output_path) {
+                    if previous.len() == 1 &&
+                        previous[0].0 == target_fn &&
+                        previous[0].1 == deps {
+                        continue
+                    }
+                }
```

Comment:
Perhaps it's time to add a 

```rust
#[derive(Eq)]
struct DepInfo { ... }

impl DepInfo {
  fn parse(r: Read) -> CargoResult<Self>;
  fn print(w: Write) -> CargoResult<Self>;
}
```

it would be nice to have printing and the corresponding parsing code alongside each other.